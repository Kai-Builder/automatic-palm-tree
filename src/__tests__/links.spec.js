import ComponentToTest from '../links';
import {Link} from 'react-router-dom';
import React from 'react';
import {shallow} from 'enzyme';

describe('Links', () => {
  const requiredProps = () => ({
  });
  const render = (props = requiredProps()) => shallow(<ComponentToTest {...props}/>);

  test('component renders', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });

  test('component has the correct links', () => {
    const expectedLinks = [
      {
        path: '/'
      },
      {
        path: '/anonymous'
      },
      {
        path: '/restricted-a'
      },
      {
        path: '/restricted-b'
      }
    ];

    const wrapper = render().find('[data-test="links"]').children();

    wrapper.forEach((children, index) => {
      const link = children.find(Link);

      expect(link.props().to).toEqual(expectedLinks[index].path);
      expect(link).toMatchSnapshot();
    });
    expect(wrapper).toMatchSnapshot();
  });
});

