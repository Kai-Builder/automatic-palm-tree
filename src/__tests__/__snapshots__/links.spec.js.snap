// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links component has the correct links 1`] = `
<Link
  replace={false}
  to="/"
>
  Home
</Link>
`;

exports[`Links component has the correct links 2`] = `
<Link
  replace={false}
  to="/anonymous"
>
  Anonymous
</Link>
`;

exports[`Links component has the correct links 3`] = `
<Link
  replace={false}
  to="/restricted-a"
>
  Restricted A
</Link>
`;

exports[`Links component has the correct links 4`] = `
<Link
  replace={false}
  to="/restricted-b"
>
  Restricted B
</Link>
`;

exports[`Links component has the correct links 5`] = `
Array [
  <li
    key="/"
  >
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
  </li>,
  <li
    key="/anonymous"
  >
    <Link
      replace={false}
      to="/anonymous"
    >
      Anonymous
    </Link>
  </li>,
  <li
    key="/restricted-a"
  >
    <Link
      replace={false}
      to="/restricted-a"
    >
      Restricted A
    </Link>
  </li>,
  <li
    key="/restricted-b"
  >
    <Link
      replace={false}
      to="/restricted-b"
    >
      Restricted B
    </Link>
  </li>,
]
`;

exports[`Links component renders 1`] = `
<ul
  className="links"
  data-test="links"
>
  <li
    key="/"
  >
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
  </li>
  <li
    key="/anonymous"
  >
    <Link
      replace={false}
      to="/anonymous"
    >
      Anonymous
    </Link>
  </li>
  <li
    key="/restricted-a"
  >
    <Link
      replace={false}
      to="/restricted-a"
    >
      Restricted A
    </Link>
  </li>
  <li
    key="/restricted-b"
  >
    <Link
      replace={false}
      to="/restricted-b"
    >
      Restricted B
    </Link>
  </li>
</ul>
`;
